models/AttendanceModel.ts

```diff
@@ -36,7 +36,14 @@ export class AttendanceModel extends BaseEntity {
       'timestamp',
       'asStaff',
     ]);
-    const publicAttendance: PublicAttendance = { ...rawAttendance };
+
+    const publicAttendance: PublicAttendance = {
+      ...rawAttendance,
+      event: {
+        ...rawAttendance.event,
+        feedback: rawAttendance.event.feedback.map((feedback) => feedback.getPublicEventFeedback()),
```

Comment:
yes it does technically leak feedback but that wouldn't be displayed anywhere since this frontend is just pulling attendance and the event. 

I'm not sure what you mean by having feedback reference attendance. I don't really see a reason for event feedback to be attached to an attendance object (other than the fact that feedback happens after attendance)? Feel free to clarify

Reply:
Feedback is a natural extension of attendance: you attend an event then maybe you provide some feedback, where feedback cannot exist without attendance. Right now, feedback is its own model with references to user/event, which implies any user can provide feedback on any event, but that's not true; any user that's attended an event can provide feedback on that specific event, so feedback should be linked to attendance not a user/event pair since that's the relation it's dependent on. Given that feedback's just a string (or an array of strings), not an object/model on its own, it could be a property of the attendance model instead of a model that references attendance to further simplify our schema.

Another strong hint I see that feedback should be linked to attendance is the added complexity in 
 `AttendanceModel::getPublicAttendance`, where getting an attendance requires an additional join from `EventModel` to `EventFeedbackModel` and another join from `EventFeedbackModel` to `UserModel` even though `UserModel`'s already part of `AttendanceModel`. And then for `EventModel`, we join to feedback by default and need to pass in `canSeeFeedback` everywhere. `findByAttendanceCode` doesn't even need feedback, and all those extra joins to our most common queries by far (find past/future events, which're prompted by every login) when the feedback objects are going to be filtered out by a `getPublic` call almost every time are going to have a noticeable impact on our performance since that's a lot of wasted database reading/computation. If feedback's tied to attendance, we don't need to do that, because anyone who can view attendance can view feedback and since it's a rarer more intentional call, we can always return feedback.

And, modeling feedback as part of attendance either makes easier or doesn't affect any feedback queries. Consider the access patterns involved with getting event feedback for a user, for an event, or for an attendance. In all those cases, it'd be much simpler to keep feedback as a property of attendance and use the existing `GET /attendance` routes. If a user wants to see all the feedback they've provided or an event host wants to see all the feedback an event got, I think it'd also add some good context to show feedback as a proportion of attendance.

Reply:
Thanks for the clear explanation, all these points make sense. 

I think my mind jumped to having event feedback tied to events because it made logical sense to me for feedback to be tied to the thing that is receiving the feedback (like movie feedback to a movie), and I never really brought attendance into the picture (which is tied to a user and event anyway). I'll restructure accordingly